### DISCUSSION TITLE
Task<T> GetJsonAsync<T> return null records when using the ".Include"

### DISCUSSION THREAD

**Discussion by wcrisafu** [oqtane/oqtane.framework#2547] [Q&A] [answered]
Hi,

I am working on a module that works perfectly when I only retrieve records from one table at the time. The same code does not work if I try to get the same records with code that uses the ".Include" (secondary table). When I use the ".Include", I can clearly see the records populated in the repository and in the controller but, when my "List<T>" it is returned to the client service module (GetJsonAsync) I get a "null" and the httpClient shows an internal error 500. To make sure make sure the issue was not the size, I retrieved two records each with two child records within. 

Any idea how to-troubleshoot?

**Comment by sbwalker** [comment]
Does your .Include() statement reference a secondary table with a 1:1 relationship to your primary table or a 1:N relationship?

**Reply by wcrisafu** [reply]
1:N relationship

**Comment by nicpitsch** [comment, marked as answer]
In my experience the solution is to set *System.Text.Json.Serialization.ReferenceHandler.IgnoreCycles* in your module Server project startup.cs.:

```c#
public class ServerStartup : IServerStartup
{
...
    public void ConfigureServices(IServiceCollection services)
    {
      // credits: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/preserve-references?pivots=dotnet-7-0#ignore-circular-references
      services.AddControllers().AddJsonOptions(options =>
      {
          // Prevent self referencing loop exception when loading related data ('Include' method in EF Core),
          // Be aware of possible issues by allowing this option (https://dotnetcoretutorials.com/2020/03/15/fixing-json-self-referencing-loop-exceptions/).
          options.JsonSerializerOptions.ReferenceHandler = System.Text.Json.Serialization.ReferenceHandler.IgnoreCycles;
      });
    }
...
}
```

This is valid for Oqtane >= 3.2.1 (replaced NewtonsoftJson by System.Text.Json).

**Reply by wcrisafu** [reply]
This worked perfectly! 
You have no idea how long I have been searching for a reason/solution. Thank you very much!
